lib/optimize/RealContentHashPlugin.js

```diff
@@ -403,6 +403,27 @@ ${referencingAssets
 			);
 		});
 	}
+
+	static hashRegExp(keys) {
+		return new RegExp(Array.from(keys, quoteMeta).join("|"), "g");
+	}
+
+	static hashReplacer(keys) {
+		return new RegExp(
+			`(\\(?)(${Array.from(keys, quoteMeta).join("|")})\\)?`,
+			"g"
+		);
+	}
+
+	static hashReplacement(hashToNewHash) {
+		return ownHashes => (ignore, caller, hash) => {
+			if (ownHashes && ownHashes.has(hash)) {
+				return "";
+			}
+			const newHash = hashToNewHash.get(hash);
+			return /[^\d]/.test(newHash) && caller ? `('${newHash}')` : newHash;
+		};
+	}
```

Comment:
We don't need to do it public, just create test case

Reply:
👌, I have to simulate a situation where we can generate an ETag with letters, rather than only numbers.